### PR TITLE
Plane: Use AHRS for groundspeed/groundcourse

### PR DESCRIPTION
**Pull request by magicrub**
Use AHRS for groundspeed/groundcourse instead of GPS

### CODE REVIEW

File: ArduPlane/servos.cpp [hunk 1]
```diff
@@ -91,7 +91,7 @@ bool Plane::suppress_throttle(void)
             millis() - started_flying_ms > MAX(launch_duration_ms, 5000U) && // been flying >5s in any mode
             adjusted_relative_altitude_cm() > 500 && // are >5m above AGL/home
             labs(ahrs.pitch_sensor) < 3000 && // not high pitch, which happens when held before launch
-            gps_movement) { // definite gps movement
+            has_ground_movement) { // definite gps movement
```
Comment (khancyr) [inline ArduPlane/servos.cpp:94]: you could remove gps here in the comment

File: libraries/AP_AHRS/AP_AHRS.h [hunk 2]
```diff
@@ -363,6 +363,16 @@ class AP_AHRS
         return groundspeed_vector().length();
     }
 
+    // return ground course estimate in degrees from 0 to 360
+    float ground_course(void) {
+        return wrap_360(degrees(groundspeed_vector().angle()));
```
Comment (khancyr) [inline libraries/AP_AHRS/AP_AHRS.h:368]: what would be the difference between this and using the yaw_sensor like in other place ? (like in https://github.com/ArduPilot/ardupilot/blob/master/libraries/GCS_MAVLink/GCS_Common.cpp#L2631)

**Review by khancyr — APPROVED** [review]
This looks good.
But it induce a breaking change as now if you don't have valid gps speed estimation you can still use the airspeed.
In case both airspeed and gps speed aren't availabe, the behavior isn't changed as ahrs will return (0,0) vector and there are only speed checks !

**Comment by rmackay9** [comment]
@khancyr in general I think for the core devs we should let them add the devCallTopic topic when they're ready to discuss in the weekly dev call.

**Review by tridge — CHANGES_REQUESTED** [review]
I'd like to see evidence of each of the changes having been thought through carefully for safety considerations

**Comment by magicrub** [comment]
I thought I flew with this change with an auto-takeoff without GPS.. but the autotakeoff code clearly requires a 3D lock so there must have been something different about my test flight. I'll check the hash in my log and report back.

@tridge has a particular concern about using ahrs instead of gps for takeoff.

**Comment by magicrub** [comment]
This PR is probably not worth the trouble to pursue. I was trying to clean things up but it got complicated so I'll just close.

**Pull request closed without merging**